modules/Installer/AWS.Tools.Installer.psm1

```diff
@@ -228,174 +245,117 @@ function Get-AvailableModuleVersion {
+            $entryStream = $entry.Open()
+            $temporaryManifestFilePath = Join-Path ([System.IO.Path]::GetTempPath()) "$([System.IO.Path]::GetRandomFileName()).psd1"
+            $manifestFileStream = [System.IO.File]::OpenWrite($temporaryManifestFilePath)
+            $entryStream.CopyTo($manifestFileStream)
+            $manifestFileStream.Close();
+
+            if ($PSVersionTable.PSVersion.Major -lt 6 -or $IsWindows) {
+                [System.Management.Automation.Signature]$manifestSignature = Get-AuthenticodeSignature -FilePath $temporaryManifestFilePath
+                if ($manifestSignature.Status -eq 'Valid' -and $manifestSignature.SignerCertificate.Subject -eq $script:AWSToolsSignatureSubject) {
+                    Write-Verbose "[$($MyInvocation.MyCommand)] Manifest signature correctly validated"
                 }
                 else {
-                    Write-Verbose "[$($MyInvocation.MyCommand)] Authenticode signature can only be vefied on Windows, skipping"
+                    throw "Error validating manifest signature for $($Name)"
                 }
+            }
+            else {
+                Write-Verbose "[$($MyInvocation.MyCommand)] Authenticode signature can only be vefied on Windows, skipping"
+            }
 
-                $manifestData = Import-PowerShellDataFile $temporaryManifestFilePath
-
-                if ($manifestData.PrivateData.MinAWSToolsInstallerVersion) {
-                    $minVersion = Get-CleanVersion $manifestData.PrivateData.MinAWSToolsInstallerVersion
-                    if ($minVersion -gt $script:CurrentMinAWSToolsInstallerVersion) {
-                        throw "$($moduleInfo.Name) version $RequiredVersion requires at least AWS.Tools.Installer version $minVersion. Run 'Update-Module AWS.Tools.Installer -Force'."
-                    }
-                }
+            [PSObject]$manifestData = Import-PowerShellDataFile $temporaryManifestFilePath
 
-                $saveAWSToolsModuleParams = @{
-                    RequiredVersion        = $RequiredVersion
-                    TemporaryRepoDirectory = $TemporaryRepoDirectory
-                    SavedModules           = $SavedModules
-                }
-                $manifestData.RequiredModules | ForEach-Object {
-                    Write-Verbose "[$($MyInvocation.MyCommand)] Found dependency $($_.ModuleName)"
-                    Save-AWSToolsModule -Name $_.ModuleName @saveAWSToolsModuleParams @proxyParams | Out-Null
+            if ($manifestData.PrivateData.ContainsKey('MinAWSToolsInstallerVersion')) {
+                [System.Version]$minVersion = Get-CleanVersion $manifestData.PrivateData.MinAWSToolsInstallerVersion
+                if ($minVersion -gt $script:CurrentMinAWSToolsInstallerVersion) {
+                    throw "$($Name) version $RequiredVersion requires at least AWS.Tools.Installer version $minVersion. Run 'Update-Module AWS.Tools.Installer -Force'."
```

Comment:
I will remove it. I added it in the first place because using `Install-Module` to perform the update would actually require `-Force` to be specified. But `Update-Module` doesn't, so I agree it is best to remove it in order to retain the confirmation prompt.